File: apps/messages/services/callmebot-api.go
package services

import (
	"fmt"
	"net/http"
	"os"
	"log"
	"strings"

	"github.com/joho/godotenv"
)

func CallMeBotAPI(text string) {
	err := godotenv.Load()
	if err != nil {
		log.Fatal("Error in loading .env file")
	}

	callMeBotAPIKey := os.Getenv("CALLMEBOT_API_KEY")
	callMeBotWhatsAppNumbers := os.Getenv("CALLMEBOT_WHATSAPP_NUMBER")


	fmt.Printf("CALLMEBOT_WHATSAPP_NUMBER: %s\n", callMeBotWhatsAppNumbers)
	if callMeBotAPIKey == "" || callMeBotWhatsAppNumbers == "" {
		fmt.Println("CALLMEBOT_API_KEY or CALLMEBOT_WHATSAPP_NUMBER is not configured. Ignoring message sending.")
		return
	}

	textReplaced := strings.ReplaceAll(text, " ", "+")
	
	url := fmt.Sprintf("https://api.callmebot.com/whatsapp.php?phone=%s&text=%s&apikey=%s",
		callMeBotWhatsAppNumbers, textReplaced, callMeBotAPIKey)

	fmt.Printf("Sending message to CallMeBot API: %s\n", url)
	resp, err := http.Get(url)
	if err != nil {
		fmt.Printf("Error sending message to CallMeBot API: %v\n", err)
		return
	}
	fmt.Printf("Response from CallMeBot API: %v\n", resp.Status)
}

File: apps/web/services/callmebot-api.ts
import { EXTERNAL_APIS_CONFIG } from "@/constants";

interface CallmebotAPIProps {
  text: string;
}

interface MessageViaCallmebotResponse {
  success: boolean;
  message: string;
}

export async function sendMessageViaCallmebotAPI({ text }: CallmebotAPIProps): Promise<MessageViaCallmebotResponse> {
  const API_KEY = process.env.CALLMEBOT_API_KEY;
  const WHATSAPP_NUMBER = process.env.CALLMEBOT_WHATSAPP_NUMBER;

  const response = await fetch(`https://api.callmebot.com/whatsapp.php?phone=${WHATSAPP_NUMBER}&text=${encodeURIComponent(text)}&apikey=${API_KEY}`, {
    method: "POST",
    next: { revalidate: EXTERNAL_APIS_CONFIG.UPDATE_INTERVAL_MS }
  });

  const responseJson = await response.json();

  return responseJson;
}

